### PR TITLE
[WIP] Template field groups

### PR DESCRIPTION
**Pull request by danielfriis**

### CODE REVIEW

File: app/views/signatures/_form.html.haml [hunk 1]
```diff
@@ -5,7 +5,7 @@
   - if signature.persisted?
     = bb_tabs do |tabs|
       - signature.template.template_fields.editable.group_by(&:template_field_group).each do |field_group, fields|
-        - tabs.tab (field_group.present? ? field_group.name : "Basics") do
+        - tabs.tab field_group.present? ? field_group.name : "Basics" do
```
Comment (kaspernj) [inline app/views/signatures/_form.html.haml]: How about `field_group&.name&.presence || "Basics"` ?

File: app/controllers/template_field_groups_controller.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
+class TemplateFieldGroupsController < ApplicationController
+  load_resource :account
+  load_resource :template, through: :account
+  load_and_authorize_resource :template_field_group, through: :template
+
+  def index
+    @ransack = @template_field_groups.ransack(params[:q])
+    @template_field_groups = @ransack.result.page(params[:q])
```
Comment (kaspernj) [inline app/controllers/template_field_groups_controller.rb]: I believe you should use `params[:page]` here

File: app/controllers/template_field_groups_controller.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  def new; end
+
+  def create
+    if @template_field_group.save
+      redirect_to [@account, @template, :template_field_groups]
+    end
```
Comment (kaspernj) [inline app/controllers/template_field_groups_controller.rb]:
```
else
  render :new
end
```

File: app/models/template_field_group.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class TemplateFieldGroup < ApplicationRecord
+  belongs_to :template
+
+  has_many :template_fields
```
Comment (kaspernj) [inline app/models/template_field_group.rb]: `has_many :template_fields, dependent: :destroy` ?

File: app/views/signatures/_form.html.haml [hunk 5]
```diff
@@ -3,13 +3,16 @@
...
-        - elsif sv.object.template_field.type == "TextTemplateField"
-          = sv.input :value, label: field.identifier.capitalize
-        = sv.hidden_field :template_field_id, value: field.id
+    = bb_tabs do |tabs|
+      - signature.template.template_fields.editable.group_by(&:template_field_group).each do |field_group, fields|
+        - tabs.tab field_group.present? ? field_group.name : "Basics" do
+          - fields.each do |field|
+            = f.simple_fields_for :signature_values, signature.signature_values.find_or_initialize_by(template_field: field) do |sv|
+              - if sv.object.template_field.type == "ImageTemplateField"
+                = sv.input :file, label: field.identifier.capitalize
```
Comment (kaspernj) [inline app/views/signatures/_form.html.haml]: Why do you define the label here? This wont be fully translatable, will it?

File: app/controllers/template_field_groups_controller.rb (hunk 3 above)
Comment (danielfriis) [inline app/controllers/template_field_groups_controller.rb]: Doh

File: app/models/template_field_group.rb (hunk 4 above)
Comment (danielfriis) [inline app/models/template_field_group.rb]: 👍

File: app/views/signatures/_form.html.haml (hunk 1 above)
Comment (danielfriis) [inline app/views/signatures/_form.html.haml]: 👍

File: app/controllers/template_field_groups_controller.rb (hunk 2 above)
Comment (danielfriis) [inline app/controllers/template_field_groups_controller.rb]: 👍

File: app/views/signatures/_form.html.haml (hunk 5 above)
Comment (danielfriis) [inline app/views/signatures/_form.html.haml]: It's an identifier used as a merge variable. Wouldn't make sense to translate :)

**Pull request merged**